src/internal/streamstore/streamstore.go

```diff
@@ -156,7 +157,7 @@ func (ss *streamStore) DeleteBackupDetails(
 ) error {
 	err := ss.kw.DeleteSnapshot(ctx, detailsID)
 	if err != nil {
-		return errors.Wrap(err, "deleting backup details failed")
+		return errors.Wrap(err, "deleting backup details")
```

Comment:
should this be using clues?

Reply:
Nope. `errors.Wrap` is perfectly fine.  And since the called function accepts `ctx`, we assume that the error creator in that func already appended structured data to the error.

Reply:
A large thrust of the design is for the two packages to be interchangeable.  I understand the want for consistency, but the goal of the package is not to replace errors.  The lack of `clues.Is()` and `clues.As()` is intentional.  It's a common expectation for packages of this kind to interface cohesively with current uses of errors and pkg/errors so that contributors aren't required to touch every point of error handling in a repo when they trade out libraries.  

Having been in this position before: my expectation is that contributors will feel less overwhelmed by code that often uses errors, and sometimes uses clues, than they would if the repo used only clues.  The latter changes the total topography of error handling (and cognitive load to understand it, even if only as an assumption, unknowns can be impenetrable).  By showcasing interoperability with errors, we can prove that the behavior remains compliant with handling that is well known and understood, at a minimum.  That provides a baseline level of comfort and eases cognitive load.

Reply:
To be clear, the following trying to put down the clues package nor is that my intention. What I'm trying to show is that there's a lot of edge cases and open questions that can pop up. Some of these I'm trying to surface here but others we may not realize until we need to work around them in implementations or worse, we see bugs appearing due to unexpected behavior

>By showcasing interoperability with errors, we can prove that the behavior remains compliant with handling that is well known and understood, at a minimum.

I think this is kind of a red herring honestly. I think what we can say is that clues is interoperable with `error` because `error` only requires `Error() string`. What we can't easily say is that the clue package is interoperable with whatever third-party error library we may be using (in our case `pkg/errors` in most places). Since we've already mostly abandoned `error` in favor of `pkg/errors`, having interoperability with just `error` is not as useful as interoperability with what we commonly use in the codebase.

If we want to be able to seamlessly use either `pkg/errors` or `clue`, what we want is for clue to be a wrapper around `pkg/error`, not to just be able to match the same `error` interface

>expectation is that contributors will feel less overwhelmed by code that often uses errors, and sometimes uses clues, than they would if the repo used only clues.

Looking at the code that's here, I'm confused by when I should use clues and when I should use errors. If I don't use clues, am I going to hamstring debugging because we don't get enough context information? If clues is not meant to be a full replacement for whatever errors package we're using, do I need to be careful about what I return? There doesn't seem to be a way to get back the wrapped/stacked errors, so what if I need context from clues but also access to some interface on the wrapped/stacked error? It seems like that functionality would become inaccessible, which makes interoperability difficult and requires more thinking about the entire call chain rather than how to handle things in an individual function. Adding handling for specific functions like `func (c *clue) Is(target error) bool` to clues is possible, but that makes clues more of a wrapper around a specific error library rather than a separate interoperable package. At best we'll get compile errors when we don't implement things. At worst we'll get silent failures because a lot of things like `As(target error)` have type assertions and just exit with false if the assertion fails

These sorts of questions start to come up as we begin integrating things, but they are really easy to overlook when just starting to work on a codebase that uses multiple packages for error handling, especially if one isn't intimately familiar with the libraries being used. If using clues doesn't give all the expected interfaces, it can easily lead to bugs because engineers may assume that it does provide all the stuff they need, especially since they're often treated as the same `error` type as other things. Yes, I get the builtin `error` type only really has `Error()` in it's interface, but once we start using other packages like `pkg/errors` the assumption kind of becomes "all errors we return have some common interface I can assume and it's the interface of the third-party package being used"